### DISCUSSION TITLE
PyAudio

### DISCUSSION THREAD

**Discussion by HamaBTW** [flet-dev/flet#4298] [Packages] [answered]
I believe PyAudio support in Flet would be incredibly beneficial. I rely on PyAudio extensively in my projects, and adding compatibility would make deploying them with Flet much smoother.

**Comment by sank7103** [comment]
did you find any alternatives? i need it too.

**Comment by ndonkoHenri** [comment, marked as answer]
Unfortunately PyAudio can't work in a Flet mobile app. PyAudio is a thin wrapper over PortAudio, and PortAudio has no host backend for iOS or Android — all of its backends are desktop (CoreAudio-macOS, ALSA, WASAPI, etc.). So even if the wheel were built, it would expose zero audio devices and provide no working audio I/O on a phone.

Concerning the alternatives, check these out:

- Recording (microphone) → [`AudioRecorder`](https://flet.dev/docs/services/audiorecorder) control
- Playback → [`Audio`](https://flet.dev/docs/services/audio) control
